blocks/form/form.js

```diff
@@ -348,6 +348,27 @@ function inputDecorator(field, element) {
   }
 }
 
+function decoratePanelContainer(container, panel) {
+  if (!container || !panel) return;
```

Comment:
Can we reduce these conditionals. If we actually need these many conditionals can we have some methods implementing the conditional logic and returning booleans. This is very difficult to understand and will be even more difficult to update/review in the future.

Also , if we have these many branching in the code, can we add extensive UTs for these. Right now this seems very messy.

Reply:
Added UTs and improved code readability.